Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -442,9 +442,9 @@ std::string MolToSmiles(const ROMol &mol, bool doIsomericSmiles, bool doKekule,
       if (rootedAtAtom == -1) {
         rootedAtAtom = std::rand() % mol.getNumAtoms();
         // need to find an atom id between 0 and mol.getNumAtoms() exclusively
-        PRECONDITION(rootedAtAtom < 0 || static_cast<unsigned int>(
-                                             rootedAtAtom) < mol.getNumAtoms(),
-                     "rootedAtomAtom must be less than the number of atoms");
+        PRECONDITION(
```

Comment:
I'm fairly sure this precondition is always true due to the mod (%)

Reply:
yep